lib/ip_str.go

```diff
@@ -2,9 +2,34 @@ package lib
 
 import (
 	"net"
+	"regexp"
 )
 
 // StrIsIPStr checks whether a string is an IP.
 func StrIsIPStr(ipStr string) bool {
 	return net.ParseIP(ipStr) != nil
 }
+
+// StrIsIPv4Str checks if the given string is an IPv4 address
+func StrIsIPv4Str(expression string) bool {
```

Comment:
We do some checking in `grepip` for this kind of stuff. You could check that and reuse/abstract-out the functionality for use in your work.